client/cnsclient/cnsclient.go

```diff
@@ -0,0 +1,84 @@
+package cnsclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/Azure/azure-container-networking/log"
+)
+
+// IpamClient specifies a client to connect to Ipam Plugin.
+type CNSClient struct {
+	connectionURL string
+}
+
+const (
+	defaultCnsURL = "http://localhost:10090"
+)
+
+// NewCnsClient create a new cns client.
+func NewCnsClient(url string) (*CNSClient, error) {
+	if url == "" {
+		url = defaultCnsURL
+	}
+
+	return &CNSClient{
+		connectionURL: url,
+	}, nil
+}
+
+// GetNetworkConfiguration Request to get network config.
+func (cnsClient *CNSClient) GetNetworkConfiguration(podName, podNamespace string) (*cns.GetNetworkContainerResponse, error) {
+	var body bytes.Buffer
+
+	httpc := &http.Client{}
+	url := cnsClient.connectionURL + cns.GetNetworkContainerByOrchestratorContext
+	log.Printf("GetNetworkConfiguration url %v", url)
+
+	podInfo := cns.KubernetesPodInfo{PodName: podName, PodNamespace: podNamespace}
+	podInfoBytes, err := json.Marshal(podInfo)
+	if err != nil {
+		log.Printf("Marshalling azure container instance info failed with %v", err)
+		return nil, err
+	}
+
+	payload := &cns.GetNetworkContainerRequest{
+		OrchestratorContext: podInfoBytes,
+	}
+
+	err = json.NewEncoder(&body).Encode(payload)
+	if err != nil {
+		log.Printf("encoding json failed with %v", err)
+		return nil, err
+	}
+
+	res, err := httpc.Post(url, "application/json", &body)
```

Comment:
The client is responsible for closing BodyReader to prevent socket leak.
i.e. 

// after checking if http.Post err is nil

defer res.Body.Close()